lintcheck/src/main.rs

```diff
@@ -116,7 +116,26 @@ impl Crate {
 
         clippy_args.extend(lint_levels_args.iter().map(String::as_str));
 
-        let mut cmd = Command::new("cargo");
+        let mut cmd;
+
+        if config.perf {
+            cmd = Command::new("perf");
+            cmd.args(&[
+                "record",
+                "-e",
+                "instructions", // Only count instructions
+                "-g",           // Enable call-graph, useful for flamegraphs and produces richer reports
+                "--quiet",      // Do not tamper with lintcheck's normal output
+                "-o",
+                "perf.data",
+                "--",
+                "cargo",
+            ]);
+            cmd.env("CARGO_PROFILE_RELEASE_DEBUG", "true");
```

Comment:
This would want to go in `build_clippy` as we want the debuginfo on the built clippy rather than the packages we're linting